### PR TITLE
#538: WIP wrap adapter code into reusable functions

### PR DESCRIPTION
**Pull request by ivanhofer**

### CODE REVIEW

File: source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { RelativeUrl } from '../../../core/index.js'
+import type { SvelteKitClientRuntime } from '../client/runtime.js'
+
+export const inlangSymbol = Symbol.for("inlang")
+
+export const setRuntimeToData = <Data extends Record<string, unknown> | null>(data: Data, runtime: SvelteKitClientRuntime): Data & { [inlangSymbol]: SvelteKitClientRuntime } =>
```
Comment (benjaminpreiss) [inline source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts:6]: how about "setRuntimeInData"?

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa/src/routes/about/+layout.ts
Comment (benjaminpreiss) [inline source-code/sdk-js/src/adapter-sveltekit/examples/spa/src/routes/about/+layout.ts:1]: Btw, why do we need to add the runtime to data? Doesn't it suffice to load it from context?

File: source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts (hunk 1 above)
Comment (benjaminpreiss) [reply source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts:6]: Or better addRuntimeToData

File: source-code/sdk-js/src/adapter-sveltekit/utils/server/state.ts [hunk 2]
```diff
@@ -15,21 +15,14 @@ export const languages = config.languages
...
-	languages: string[]
-	language: string
-	i: InlangFunction
-}
-
 export const getResource = (language: string) =>
 	resources.find(({ languageTag: { name } }) => name === language)
 
-export const setInlangInformationToLocals = (
+export const setRuntimeToLocals = (
```
[the remark this answers is not in the document]
Comment (benjaminpreiss) [reply source-code/sdk-js/src/adapter-sveltekit/utils/server/state.ts:21]: I would say we name this addRuntimeToLocals

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/inlang/[language].json/+server.ts
Comment (benjaminpreiss) [inline source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/inlang/[language].json/+server.ts:1]: Why do we need this proxy endpoint?

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/+page.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import { inlangSymbol } from "../inlang.js"
+import { getRuntimeFromData } from "@inlang/sdk-js/adapter-sveltekit/shared"
 import type { PageLoad } from "./$types.js"
 
 export const load = (async ({ parent }) => {
 	if (browser) {
 		const data = await parent()
 
-		const i = data[inlangSymbol]!.getInlangFunction()
+		const i = getRuntimeFromData(data).getInlangFunction()
```
Comment (benjaminpreiss) [inline source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/+page.ts:9]: Why do we need to specifically call `.getInlangFunction()` here? Is it possible to offer `i` as a property:

`const {i} = getRuntimeFromData(data)`

File: source-code/sdk-js/src/adapter-sveltekit/utils/client/runtime.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export const initSvelteKitClientRuntime = async ({
+	fetch,
+	language,
+	referenceLanguage,
+	languages,
+}: InitSvelteKitClientRuntimeArgs) => {
+	const runtime = initRuntime({
+		readResource: async (language: string) =>
+			fetch(`/inlang/${language}.json`).then((response) =>
```
[the remark this answers is not in the document]
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/utils/client/runtime.ts:19]: This is the `SvelteKit` endpoint where we can load Resources from the server. The server uses the `inlang.config.js` file

File: source-code/sdk-js/src/adapter-sveltekit/utils/client/reactive/index.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	i: Readable<InlangFunction>
+		switchLanguage: (language: Language) => Promise<void>
+	loadResource: SvelteKitClientRuntime["loadResource"]
+	route: (href: RelativeUrl) => RelativeUrl
+}
+
+export const getRuntimeFromContext = (): RuntimeContext => getContext(inlangSymbol)
+
+export const setRuntimeToContext = (runtime: SvelteKitClientRuntime) => {
+	const _language = writable(runtime.getLanguage() as string)
```
[the remark this answers is not in the document]
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/utils/client/reactive/index.ts:31]: Svelte stores can also be used in any other framework.

File: source-code/sdk-js/src/adapter-sveltekit/utils/server/state.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/utils/server/state.ts:21]: the `locals` object that get's passed to all server functions

File: source-code/sdk-js/src/adapter-sveltekit/utils/client/not-reactive/index.ts
[the remark this answers is not in the document]
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/utils/client/not-reactive/index.ts:1]: Then we won't be able to use three shaking.

File: source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts (hunk 1 above)
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/utils/shared/utils.ts:6]: I was also thinking of renaming that function

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa/src/routes/about/+layout.ts
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/examples/spa/src/routes/about/+layout.ts:1]: Contexts can't be used outside the Svelte rendering process

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/inlang/[language].json/+server.ts
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/inlang/[language].json/+server.ts:1]: to load the resources on the client

File: source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/+page.ts (hunk 3 above)
Comment (ivanhofer) [reply source-code/sdk-js/src/adapter-sveltekit/examples/spa-static/src/routes/+page.ts:9]: This is one of the refactorings I still need to finish ;)

File: source-code/sdk-js/src/adapter-sveltekit/utils/client/reactive/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (samuelstroschein) [reply source-code/sdk-js/src/adapter-sveltekit/utils/client/reactive/index.ts:31]: >  @samuelstroschein You said, that solid offers framework-agnostic reactivity? Can you point me to where you have this information from?

I was referring to the fact that SolidJS works in plain JS, see https://www.youtube.com/watch?v=J70HXl1KhWE. But, using the framework's reactive primitives seems to make more sense. First, no unexpected reactivity issues, and second, the bundle size becomes 0kb :)

**Pull request merged**